component/loki/source/cloudflare/types.go

```diff
@@ -0,0 +1,63 @@
+package cloudflare
+
+import (
+	"fmt"
+	"time"
+
+	cft "github.com/grafana/agent/component/loki/source/cloudflare/internal/cloudflaretarget"
+	"github.com/grafana/agent/pkg/river"
+	"github.com/prometheus/common/model"
+)
+
+// Config defines how to create a Cloudflare logs target.
+type Config struct {
+	APIToken   string            `river:"api_token,attr"`
+	ZoneID     string            `river:"zone_id,attr"`
+	Labels     map[string]string `river:"labels,attr,optional"`
+	Workers    int               `river:"workers,attr,optional"`
+	PullRange  time.Duration     `river:"pull_range,attr,optional"`
+	FieldsType string            `river:"fields_type,attr,optional"`
```

Comment:
It feels a bit weird for the bulk of the config to be in a `config` block. Was there a reason to do this vs keeping a flat list of attributes in the arguments?

Reply:
No, let's go with the flatter Arguments structure.